### PR TITLE
Fixed Unable to Open User Home & Added Complete test coverage for Home

### LINKED ISSUE #1651: Bug Report: Unable to open User joined Organization Home Screen due to Error
**Describe the bug**
- If a user opens organization's home screen which he has joined error is shown. 
- The error can be traced to following file `src/screen/UserPortal/Home/Home.tsx` which occurs due to accessing undefined attribute.

**To Reproduce**

1. Sign in as user (Must've joined atleast 1 org)
2. Open any joined Organizations home page
3. The error will be shown

**Actual behavior**
An error occurs due to accessing undefined attribute without proper checks

**Screenshots**

https://github.com/PalisadoesFoundation/talawa-admin/assets/109027247/d6422cbe-2807-4a66-87b0-5f26f8377aab

**Expected behavior** (Have identified and solved the bug on local)
No Error must be shown. and Home Screen must be displayed 

https://github.com/PalisadoesFoundation/talawa-admin/assets/109027247/37f9be97-bc09-4138-ad7c-6fd9fec87cd1

Additionally write tests for `src/screen/UserPortal/Home/Home.tsx` whose code coverage currently 
[Link](https://app.codecov.io/gh/PalisadoesFoundation/talawa-admin/blob/develop/src%2Fscreens%2FUserPortal%2FHome%2FHome.tsx) 
image

### PR DESCRIPTION
**Pull request by GlenDsza**
**What kind of change does this PR introduce?**

bugfix,  Adding tests for complete test coverage, refactoring, 

**Issue Number:**

Fixes #1651  and #1131 (both issues were extension of each other)

**Did you add tests for your changes?**

Yes 

**Snapshots/Videos:**

https://github.com/PalisadoesFoundation/talawa-admin/assets/109027247/bde6fe56-3fab-4186-bf16-995a0a142ccc

Test Coverage for `Home.tsx`:
image

Test Coverage for `StartPostModal.tsx`:
image

**Summary**

- Added check & fixed typo in accessing attribute for setting posts which was leading to error mentioned in #1651 
- Refactored `src/screens/UserPortal/Home/Home.tsx` for readability 
- Shifted SVG constants code from Home.tsx to `src/assets/svgs`
- Shifted `InterfacePostCard` to `src/utils/interface.ts` to reduce code duplication as it was being used in multiple file 
- Improved modularity of code by Shifting Modal to create post into separate `StartPostModal.tsx` in src/components/UserPortal/StartPostModal.tsx. wrote respective test cases. 
-  Added tests in `Home.tsx` for 100% code coverage as per #1131 

**Does this PR introduce a breaking change?**

No

**Other information**

**Have you read the [contributing guide](https://github.com/PalisadoesFoundation/talawa-admin/blob/master/CONTRIBUTING.md)?**

Yes

### CODE REVIEW

**Comment by GlenDsza** [comment]
@palisadoes Need PR Reviewers

**Comment by Huy1996** [comment]
@beingnoble03 @aashimawadhwa Can you review this PR?

File: src/screens/UserPortal/Home/Home.tsx [hunk 1]
```diff
@@ -324,174 +188,91 @@ export default function home(): JSX.Element {
...
+                  title,
+                  text,
+                  likeCount,
+                  commentCount,
+                } = node;
 
-                const postComments: any = [];
-                post.comments.forEach((value: any) => {
-                  const commentLikes: any = [];
+                // const allLikes: any =
```
Comment (GlenDsza) [inline src/screens/UserPortal/Home/Home.tsx:212]: Commented these lines due to missing likedBy mentioned in #1688

**Comment by GlenDsza** [comment]
Few lines for rendering of posts in PostCard related to `likedBy` and `comments` in `src/screens/UserPortal/Home/Home.tsx` were commented out due missing of likedBy and comments in `ORGANIZATION_POST_LIST` mentioned in #1688

**Comment by aashimawadhwa** [comment]
All functionalities looks good to me, this can be merged further. @GlenDsza @Huy1996

**Review by aashimawadhwa — APPROVED** [review]

**Review by palisadoes — APPROVED** [review]

**Pull request merged**